utils/build/docker/golang/app/net-http/main.go

```diff
@@ -626,6 +626,30 @@ func main() {
 		appsec.TrackUserLoginSuccessEvent(r.Context(), user, map[string]string{}, tracer.WithUserSessionID(cookie.Value))
 	})
 
+	mux.HandleFunc("/inferred-proxy/span-creation", func(w http.ResponseWriter, r *http.Request) {
+		statusCodeStr := r.URL.Query().Get("status_code")
+		statusCode := 200
+		if statusCodeStr != "" {
+			var err error
+			statusCode, err = strconv.Atoi(statusCodeStr)
+			if err != nil {
+				statusCode = 400 // Bad request if conversion fails
+			}
+		}
+
+		// Log the request headers
+		fmt.Println("Received an API Gateway request")
+		for key, values := range r.Header {
+			for _, value := range values {
+				fmt.Printf("%s: %s\n", key, value)
+			}
+		}
+
```

Comment:
Any thoughts on moving these lines of code to a shared file (i.e. in the `internal` folder)? There's a lot of repetitive code that might be better suited under a separate function with a descriptive name. But I'll defer to the @DataDog/system-tests-core team for final thoughts.

Reply:
There are tiny differences between each web server that prevent us from generalizing it as much as we would want to. For now, I think we can keep it like this and perhaps look at a greater refactor down the road to generalize for all webservers.